server.go

```diff
@@ -606,6 +606,7 @@ func (r *oauthProxy) createUpstreamProxy(upstream *url.URL) error {
 	// create the forwarding proxy
 	proxy := goproxy.NewProxyHttpServer()
 	proxy.Logger = httplog.New(ioutil.Discard, "", 0)
+	proxy.KeepDestinationHeaders = true
```

Comment:
What is the connection between the intent of this PR and this change? What this specifically the issue that needed to be fixed here?

Reply:
@fredbi I did not write this code, but I expect this is intended to fix this issue described in [KEYCLOAK-11077](https://issues.jboss.org/browse/KEYCLOAK-11077)

> ... the access token was refreshed but the `kc-access` cookie was not updated. Subsequent requests would then not find the refresh token corresponding to the outdated `kc-access` cookie, because it has been deleted from the store after the refresh. So the user is again redirected to the authorization endpoint.
This bug is also present when stores are not used, but is less visible because the refresh token is passed along in the request. In that case each request triggers a refresh after the initial access token has expired, until the initial refresh token also expires.
**Root Cause**: goproxy removes by default all HTTP headers that have been set in previous middlewares. The KeepDestinationHeaders option can be used to prevent this.

Reply:
@fredbi yes, the issue is that without this setting, `goproxy` removes all headers that have been previously set, including the new `kc-access` cookie that should be set after token refresh.

Reply:
ok got it. Thanks. LGTM